impala/dbapi/__init__.py

```diff
@@ -27,29 +27,64 @@
 from impala.error import (Error, Warning, InterfaceError, DatabaseError,
                           InternalError, OperationalError, ProgrammingError,
                           IntegrityError, DataError, NotSupportedError)
-from impala.util import warn_deprecate_hs2
+from impala.util import warn_deprecate_hs2, warn_deprecate
+
+
+AUTH_MECHANISMS = ['NOSASL', 'PLAIN', 'GSSAPI', 'LDAP']
+
 
 # PEP 249 module globals
 apilevel = '2.0'
 threadsafety = 1  # Threads may share the module, but not connections
 paramstyle = 'pyformat'
 
+def connect(host='localhost', port=21050, protocol='hiveserver2', database=None,
+            use_ldap=None, ldap_user=None, ldap_password=None,  # deprecated
+            use_kerberos=None,  # deprecated
+            auth_mechanism=None, **kwargs):
+
+    if auth_mechanism:
+        auth_mechanism = auth_mechanism.upper()
+    else:  # 'NOSASL' is default
+        auth_mechanism = 'NOSASL'
+
+    if auth_mechanism not in AUTH_MECHANISMS:
+        raise NotSupportedError(
+            'Unsupported authentication mechanism: {0}'.format(auth_mechanism))
+
+    if use_kerberos is not None:
+        warn_deprecate('use_kerberos', 'auth_mechanism="GSSAPI"')
+        if use_kerberos:
+            if auth_mechanism != 'GSSAPI':
+                raise InterfaceError("Kerberos requires authentication mechanism 'GSSAPI'")
+            else:
+                auth_mechanism = 'GSSAPI'
```

Comment:
I think we need to reverse the inner if/else:

```python
if auth_mechanism is None:
    auth_mechanism = 'GSSAPI'
elif auth_mechanism != 'GSSAPI':
    # inconsistent information
    raise ...
```

Otherwise, leaving `auth_mechanism=None` will raise an `InterfaceError`.